test/prepare_disk.sh

```diff
@@ -0,0 +1,254 @@
+
+ARM_EMU_BIN="${ARM_EMU_BIN:-}"
+
+CWD="$(pwd)"
+
+SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
+PARTITION_TABLE_FILE="test_jedi_emmc_sfdisk.table"
+TARGET_STORAGE_DEVICE=""
+
+JEDI_PARTITION_TABLE_FILE_NAME="config/jedi_emmc_sfdisk.table"
+PREPARE_DISK_COMMAND="/etc/system_update.d/30_prepare_disk.sh"
+
+STORAGE_DEVICE_IMG="storage_device.img"
+BYTES_PER_SECTOR="512"
+STORAGE_DEVICE_SIZE="7553024" # sectors, about 3.6 GiB
+
+TEST_OUTPUT_FILE="$(mktemp -d)/test_results_$(basename "${0%.sh}").txt"
+
+toolbox_image=""
+toolbox_root_dir=""
+work_dir=""
+
+exit_on_failure=false
+result=0
+
+
+execute_prepare_disk()
+{
+    chroot_environment="SYSTEM_UPDATE_DIR=${SYSTEM_UPDATE_DIR}"
+    chroot_environment="${chroot_environment} PARTITION_TABLE_FILE=${PARTITION_TABLE_FILE}"
+    chroot_environment="${chroot_environment} TARGET_STORAGE_DEVICE=${TARGET_STORAGE_DEVICE}"
+
+    sha512sum "${PARTITION_TABLE_FILE}" > "${PARTITION_TABLE_FILE}.sha512"
+    cp "${PARTITION_TABLE_FILE}" "${toolbox_root_dir}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}"
+    cp "${PARTITION_TABLE_FILE}.sha512" "${toolbox_root_dir}${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}.sha512"
+
+    chroot "${toolbox_root_dir}" /bin/sh -c "${chroot_environment} ${PREPARE_DISK_COMMAND}" || return 1
+
+    sfdisk -d "${TARGET_STORAGE_DEVICE}" > "${PARTITION_TABLE_FILE}.verify"
+
+    # Remove the identifiers in the header because they will always change.
+    sed -i "s/label-id:.*//" "${PARTITION_TABLE_FILE}"
+    sed -i "s/label-id:.*//" "${PARTITION_TABLE_FILE}.verify"
+
+    # Remove the name from the source partition file, as the names are not stored in the partition table.
+    sed -i "s/, name=.*//" "${PARTITION_TABLE_FILE}"
+
+    diff -b "${PARTITION_TABLE_FILE}" "${PARTITION_TABLE_FILE}.verify" || return 1
+}
+
+test_disk_integrity()
+{
+    # All return codes not 0 should be considered an error, since prepare_disk
+    # should have fixed any potential filesystem error.
+    sfdisk -Vl "${TARGET_STORAGE_DEVICE}"
+    fsck.ext4 -fn "${TARGET_STORAGE_DEVICE}p1"
+    fsck.f2fs "${TARGET_STORAGE_DEVICE}p2"
+    fsck.f2fs "${TARGET_STORAGE_DEVICE}p3"
+}
+
+create_dummy_storage_device()
+{
+    echo "Creating test image: '${STORAGE_DEVICE_IMG}'."
+
+    dd if="/dev/zero" of="${STORAGE_DEVICE_IMG}" bs="1" count="0" \
+        seek="$((BYTES_PER_SECTOR * STORAGE_DEVICE_SIZE))"
+
+    echo "writing partition table:"
+
+    sfdisk "${STORAGE_DEVICE_IMG}" < "${CWD}/${JEDI_PARTITION_TABLE_FILE_NAME}"
+
+    echo "formatting partitions"
+
+    TARGET_STORAGE_DEVICE="$(losetup --show --find --partscan "${STORAGE_DEVICE_IMG}")"
+
+    mkfs.ext4 -q "${TARGET_STORAGE_DEVICE}p1"
+    mkfs.f2fs -q "${TARGET_STORAGE_DEVICE}p2"
+    mkfs.f2fs -q "${TARGET_STORAGE_DEVICE}p3"
+
+    test_disk_integrity
+
+    echo "Successfully created dummy storage device: '${TARGET_STORAGE_DEVICE}'."
+}
+
+setup()
+{
+    toolbox_root_dir="$(mktemp -d -t "um-update-toolbox.XXXXXXX")"
```

Comment:
why did you not use the name template that you used with the update_files script? would have been nice and consistent :p